applicationset/generators/duck_type_test.go

```diff
@@ -290,9 +294,19 @@ func TestGenerateParamsForDuckType(t *testing.T) {
 				Resource: "ducks",
 			}: "DuckList"}
 
-			fakeDynClient := dynfake.NewSimpleDynamicClientWithCustomListKinds(runtime.NewScheme(), gvrToListKind, testCase.resource)
+			fakeDynClient := fake.NewSimpleDynamicClientWithCustomListKinds(runtime.NewScheme(), gvrToListKind, testCase.resource)
 
-			duckTypeGenerator := NewDuckTypeGenerator(t.Context(), fakeDynClient, appClientset, "namespace")
+			clusterInformer, err := settings.NewClusterInformer(appClientset, "namespace")
+			require.NoError(t, err)
+
+			ctx, cancel := context.WithCancel(t.Context())
+			defer cancel()
+			go clusterInformer.Run(ctx.Done())
+			if !cache.WaitForCacheSync(ctx.Done(), clusterInformer.HasSynced) {
+				t.Fatal("Timed out waiting for caches to sync")
+			}
```

Comment:
This boilerplate is always the same. Can you extract it to a utility function so the usage is more straightforward

```go
clusterInformer, cancel := getTestClusterInformer(t, clientSet)
defer cancel()
```